src/Components/ServiceScene/LogsTableScene.tsx

```diff
@@ -34,7 +39,68 @@ export class LogsTableScene extends SceneObjectBase<LogsTableSceneState> {
     this.setState({
       menu: new PanelMenu({ addInvestigationsLink: false }),
     });
+    this.onActivateSyncDisplayedFieldsWithUrlColumns();
+  }
+
+  private getParentScene() {
+    return sceneGraph.getAncestor(this, LogsListScene);
   }
+
+  // on activate sync displayed fields with url columns
+  onActivateSyncDisplayedFieldsWithUrlColumns = () => {
+    const searchParams = new URLSearchParams(locationService.getLocation().search);
+    const urlColumnsParam = searchParams.get('urlColumns');
+    const urlColumnsUrl = urlColumnsParam ? (JSON.parse(decodeURIComponent(urlColumnsParam)) as string[]) : null;
```

Comment:
This seems like a good candidate to use https://github.com/grafana/logs-drilldown/blob/main/src/services/narrowing.ts that was introduced by @gtk-grafana to have more confidence on the actual type of stuff like user input, parsed json, etc.
We could add a narrowArray(type: string) or narrowStringArray().

Reply:
Updated! Interesting concept, I took a look they all kind of seem like type validations. I wonder about naming them isValidType or isEmpty.